### PR TITLE
Add annotations to the operation - case permission validation - Fix 12219 15822

### LINKED ISSUE #12219: [REQ][JAVA][SPRING] Support for auto generating spring security annotations in controllers/methods
### Is your feature request related to a problem? Please describe.

This feature request is related to the desire to keep as much of my API configuration generated through the Open API spec file as possible. Currently if I want to set up authentication, I can define some of what I want inside my YML file but I still need to manually go through and do security setup on the controller/request level.

## Describe the solution you'd like

Introduction of an Open API extension for `securitySchemes` (e.g. `x-spring-preauthorize`) whose value is a string. When a scheme using this extension is used on a request, the [Spring `@PreAuthorize(...)`](https://www.baeldung.com/spring-security-method-security#3-using-preauthorize-and-postauthorize-annotations) annotation is applied to the method in the `*Api` class. If a scheme using this extension is used in the root level `security` property, my understanding is that the scheme would automatically be applied to each operation individually unless it's overridden on the request level.

## Describe alternatives you've considered

- Alternatives to `@PreAuthorize` tag but still auto-generating: `@Secured`, `@RolesAllowed`
  - `@PreAuthorize` provides the best flexibility with implementation including the ability to specify different security requirements besides roles, and being able to AND and OR different requirements (e.g. `hasRole("ROLE_A") AND hasRole("ROLE_B")`).
- Alternative to auto generating: manual configuration
  - Manual configuration is possible and feasible when extending the `*Api` class and adding the additional annotation in the controller. However, this decouples the authorization from the spec document for my API, so ideally I would prefer to have it all working together.

### Example YAML and Generated Method

YAML Config (Lines omitted for brevity)

```yaml
paths:
  /pet:
    put:
      tags:
      - pet
      summary: Update an existing pet
      security:
      - spring_role_auth
components:
  securitySchemes:
    spring_role_auth:
      type: http
      scheme: basic
      x-spring-preauthorize: "'hasRole(ROLE_A') AND hasRole('ROLE_B')"
```

Generated API File Method

```java
//Added import
import org.springframework.security.access.prepost.PreAuthorize;

//New annotation and where it could go in the annotation order
@RequestMapping()
@PreAuthorize("hasRole('ROLE_A') AND hasRole('ROLE_B')")
default ResponseEntity<Pet> addPet() {}
```

### Final Note

I might have the time to also work on a pull request to implement this, but wanted to submit this issue to get feedback on the idea first before anything is started. Thanks!

### LINKED ISSUE #15822: [BUG] Description securityScheme ignored and not creating RolesAllowed annotation
#### Bug Report Checklist

- [ ] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [ ] Have you searched for related issues/PRs?
- [ ] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

I would like to use the @RolesAllowed annotation for my endpoints to specify the role that is allowed. I'm generating my API interfaces from an openapi.yml and I added

```
     security:
      - SecurityScheme:
        - user
```
Now I openapi-generator to generate the java API but no @RolesAllowed annotation is added.

### PR DESCRIPTION
**Pull request by rodrigoma3**
Add annotations to the operation. There are 3 annotations for permission validation and they all have their advantages, and they need to be in operation to work, so it's open to add any annotation (identical to x-field-extra-annotation and x-class-extra-annotation). 
fix #12219 
fix #15822 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.
@cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09)

### CODE REVIEW

**Comment by rodrigoma3** [comment]
@wing328 Could you approve this PR?

File: modules/openapi-generator/src/test/resources/2_0/issue12219.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        - in: body
+          name: Obj to test
+          schema:
+            $ref: '#/definitions/ObjTest'
+      responses:
+        '201':
+          description: successful operation
+          schema:
+            $ref: '#/definitions/ObjTest'
+      x-operation-extra-annotation: '@javax.annotation.security.RolesAllowed({"ROLE_TEST"}) @org.springframework.security.access.annotation.Secured({"ROLE_TEST"}) @org.springframework.security.access.prepost.PreAuthorize("hasRole(''ROLE_TEST'') and hasRole(''ROLE_TEST_TWO'')")'
```
Comment (borsch) [inline modules/openapi-generator/src/test/resources/2_0/issue12219.yaml:31]: add onle more test with array like style(which I believe would be used widely)

```yaml
x-operation-extra-annotation: 
  - @javax.annotation.security.RolesAllowed({"ROLE_TEST"})
  - @org.springframework.security.access.annotation.Secured({"ROLE_TEST"})
  - @org.springframework.security.access.prepost.PreAuthorize("hasRole('ROLE_TEST') and hasRole('ROLE_TEST_TWO')")
```

**Review by borsch — CHANGES_REQUESTED** [review]

File: modules/openapi-generator/src/test/resources/2_0/issue12219.yaml (hunk 1 above)
Comment (rodrigoma3) [reply modules/openapi-generator/src/test/resources/2_0/issue12219.yaml:31]: done

**Review by borsch — APPROVED** [review]

**Comment by borsch** [comment]
@rodrigoma3 please update samples and I'll merge this PR

**Comment by rodrigoma3** [comment]
> @rodrigoma3 please update samples and I'll merge this PR

done

**Pull request merged**